src/test/common/get_command_descriptions.cc

```diff
@@ -97,8 +97,7 @@ static void pull585()
 }
 
 int main(int argc, char **argv) {
-  vector<const char*> args;
-  argv_to_vec(argc, (const char **)argv, args);
+  auto args = argv_to_vec(argc, argv);
```

Comment:
```
../src/test/common/get_command_descriptions.cc:100:15: error: no matching function for call to 'argv_to_vec'
  auto args = argv_to_vec(argc, argv);
              ^~~~~~~~~~~
../src/common/ceph_argparse.h:49:33: note: candidate function not viable: no known conversion from 'char **' to 'const char **' for 2nd argument
extern std::vector<const char*> argv_to_vec(int argc, const char** argv);
                                ^
```

do we need a variant which accepts `char **` well?

Reply:
I made a mistake when reordering the commits (that's why I didn't yet remove the draft status).
Almost finished cleaning up now.